ui/appui-react/src/appui-react/ui-items-provider/StandardContentToolsUiItemsProvider.tsx

```diff
@@ -0,0 +1,137 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module StandardUiItemsProvider
+ */
+
+import * as React from "react";
+import { ViewClipByPlaneTool } from "@itwin/core-frontend";
+import {
+  CommonStatusBarItem, CommonToolbarItem,
+  StatusBarSection, ToolbarOrientation, ToolbarUsage, UiItemProviderOverrides,
+  UiItemsManager, UiItemsProvider,
+} from "@itwin/appui-abstract";
+import { SelectionContextToolDefinitions } from "../selection/SelectionContextItemDef";
+import { StatusBarItemUtilities } from "../statusbar/StatusBarItemUtilities";
+import { withStatusFieldProps } from "../statusbar/withStatusFieldProps";
+import { SectionsStatusField } from "../statusfields/SectionsField";
+import { ToolbarHelper } from "../toolbar/ToolbarHelper";
+import { CoreTools } from "../tools/CoreToolDefinitions";
+
+/**
+ * Defines what tools to include from the provider. If any tools in the horizontal or vertical group are
+ * specified then only those tools will be provided to stage.
+ * @public
+ */
+export interface DefaultContentTools {
+  horizontal?: {
+    clearSelection?: boolean;
+    clearDisplayOverrides?: boolean;
+    /** if group then group button is shown to allow user to hide by element, category, model */
+    hide?: "group" | "element";
+    /** if group then group button is shown to allow user to isolate by element, category, model */
+    isolate?: "group" | "element";
+    /** only element is currently support for emphasize */
+    emphasize?: "element";
+  };
+  vertical?: {
+    selectElement?: boolean;
+    measureGroup?: boolean;
+    sectionGroup?: boolean;
+  };
+}
+
+function getGroupPriority(potentialId: any, defaultValue: number) {
+  if (undefined === potentialId)
+    return defaultValue;
+
+  // istanbul ignore else
+  if (typeof potentialId === "number") {
+    return potentialId;
+  }
+
+  // istanbul ignore next
+  return defaultValue;
+}
+
+/**
+ * @beta
+ */
+export class StandardContentToolsUiItemsProvider implements UiItemsProvider {
+  public get id(): string { return "appui-react:StandardContentToolsUiItemsProvider"; }
+
+  public static register(defaultContentTools?: DefaultContentTools, overrides?: UiItemProviderOverrides) {
```

Comment:
do we really need a static method to register for all of these classes that implement the interface?

Reply:
Should that really be apart of the UiItemsProvider ?

I worry the use of statics hear will cause issues with different core versions

Reply:
@aruniverse  - I just pushed updates to remove the static register methods.

Reply:
thx, looks better. Providers registering tools themselves is a question i have, but ill save it for later :)